pyhap/accessory.py

```diff
@@ -109,27 +109,46 @@ def pincode(self):
             self._pincode = util.generate_pincode()
         return self._pincode
 
-    def _set_services(self):
-        """Sets the services for this accessory.
+    def add_info_service(self):
+        """Helper method to add the required `AccessoryInformation` service.
 
-        The default implementation adds only the AccessoryInformation services
-        and sets its Name characteristic to the Accessory's display name.
-
-        .. note:: When inheriting from Accessory and overriding this method,
-            always call the base implementation first, as it reserves IID of
-            1 for the Accessory Information service (HAP requirement).
+        Called in `__init__` to be sure that it is the first service added.
+        May be overridden.
         """
-        info_service = get_serv_loader().get_service("AccessoryInformation")
-        info_service.get_characteristic("Name")\
-                    .set_value(self.display_name, False)
-        info_service.get_characteristic("Manufacturer")\
-                    .set_value("Default-Manufacturer", False)
-        info_service.get_characteristic("Model")\
-                    .set_value("Default-Model", False)
-        info_service.get_characteristic("SerialNumber")\
-                    .set_value("Default-SerialNumber", False)
-        # FIXME: Need to ensure AccessoryInformation is with IID 1.
-        self.add_service(info_service)
+        serv_info = get_serv_loader().get_service('AccessoryInformation')
+        serv_info.configure_char('Name', value=self.display_name)
+        serv_info.configure_char('SerialNumber', value='default')
+        self.add_service(serv_info)
+
+    def set_info_service(self, firmware_revision=None, manufacturer=None,
+                         model=None, serial_number=None):
+        """Quick assign basic accessory information."""
+        serv_info = self.get_service('AccessoryInformation')
+        if firmware_revision:
+            serv_info.configure_char(
+                'FirmwareRevision', value=firmware_revision)
+        if manufacturer:
+            serv_info.configure_char('Manufacturer', value=manufacturer)
+        if model:
+            serv_info.configure_char('Model', value=model)
+        if serial_number:
+            if len(serial_number) >= 1:
+                serv_info.configure_char('SerialNumber', value=serial_number)
+            else:
+                logger.warning(
+                    "Couldn't add SerialNumber for %s. The SerialNumber must "
+                    "be at least one character long.", self.display_name)
+
+    def add_preload_service(self, service, chars=None):
+        """Define and return a service to be available for the accessory."""
+        service = get_serv_loader().get_service(service)
+        if chars:
+            chars = chars if isinstance(chars, list) else [chars]
```

Comment:
Let's expect a single type here (a list)

Reply:
I would like to leave it in there and except a single char as string, since there are actually quite a lot of situations where you want to add **only one** other char.